### ISSUE TITLE
How to set UID / GUI

### ISSUE THREAD

**Issue by RaymondMouthaan** [grafana/grafana-docker#155]
Hai all,

As a great fan of Grafana (running on raspberry pi's), i like to know how to set a custom uid / gid of the grafana user within the container, without building an new image with --build-arg GF_UID=xxx --build-arg GF_GID=yyy.

Read from the docs (http://docs.grafana.org/installation/docker/#migration-from-a-previous-version-of-the-docker-container-to-5-1-or-later) and Dockerfile the default UID = 472 and can be overridden by the --user argument to the docker run command.

When running this `docker run -it --user 9999 --name mytest fg2it/grafana-armhf:v5.1.0` only the database file has been set to 9999 as uid, but the uid/gid of the grafana user remains 472.

```
root@25cb8f1b583b:/# ls -al /var/lib/
...
drwxrwxrwx 1 grafana grafana 4096 Apr 30 09:15 grafana
...
root@25cb8f1b583b:/# ls -al /var/lib/grafana/
drwxrwxrwx 1 grafana grafana   4096 Apr 30 09:15 .
drwxr-xr-x 1 root    root      4096 Apr 30 08:40 ..
-rw-r--r-- 1    9999 root    393216 Apr 30 09:15 grafana.db
drwxrwxrwx 2 grafana grafana   4096 Apr 30 08:40 plugins
root@25cb8f1b583b:/# id grafana
uid=472(grafana) gid=472(grafana) groups=472(grafana)
root@25cb8f1b583b:/#
```

Here I was excepting to set the the uid (and possibly the gid) of grafana and the owner of /var/lib/grafana/ to remain grafana. When this would be, one is able to set the uid/gid of the grafana user within the docker container equal to a uid/gid of the host. This would make volume persistence possible without conflicting user permissions.

Greets,
Ray

**Comment by RaymondMouthaan** [comment]
fg2it/grafana-on-raspberry#37

**Comment by DanCech** [comment]
That is how it's supposed to work.

Within the container the grafana user is always going to be 472, when you pass the `--user` parameter that just means that `grafana-server` will be run as that user id, which is why the database file is created with the `9999` uid.  The reason for specifying the user that grafana-server should run as is so that it can use the same uid that you want the mounted files to be owned as.

**Comment by RaymondMouthaan** [comment]
Yes, I understand that, but why is not the grafana root directory (/var/lib/grafana) and all things within set to this custom user? 

What I mean is, support the host persistent directory has been set to 1000:1000 then using the --user 1000 option sets the database accordingly, but the plugins dir remains de default uid 472. This gives errors while creating or modifying it. 

Or do I miss something here, please suggest which steps to take.

**Comment by DanCech** [comment]
@RaymondMouthaan Grafana doesn't change the permissions or ownership of any of the files, if you want them to be owned by a particular user the suggested approach is to use a volume mounted at `/var/lib/grafana`

**Comment by RaymondMouthaan** [comment]
@DanCech, currently I am not at home to try it out, but when I do I will do some tests regarding persistence. Before I faced issues regarding permissions. Maybe they are solved knowing a little more about uid and how grafana deals with it. 

Ill post the results asap :)

Thanks for your replies 👍🏼.
